src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs

```diff
@@ -469,7 +469,14 @@ private void SendPackets(SocketImplementationType type, SendPacketsElement[] ele
                     if (contentExpected != null) {
                         // test server should just echo back, so read length expected bytes from the stream
                         var contentActual = new byte[bytesExpected];
-                        var bytesReceived = sock.Receive(contentActual);
+                        var bytesReceived = 0;
+                        var tries = 0;
+                        const int MaxTries = 15;
+                        while (bytesReceived < bytesExpected && tries < NaxTries) {
+                            bytesReceived += sock.Receive(contentActual, bytesReceived, bytesExpected, SocketFlags.None);
```

Comment:
There's no guarantee that Receive returns anything larger than 1.  Why the 15 limit?

Reply:
It's absolutely arbitrary. Just trying to limit the number of receives and test duration to a reasonable time before declaring failure. In practice the tests don't exceed this limit, and I reckon it's pretty unlikely to given they all execute locally.

All that said, happy to remove the `MaxTries` altogether if you prefer.